pkg/virt-handler/device-manager/mediated_devices_types.go

```diff
@@ -75,16 +75,18 @@ func (m *MDEVTypesManager) getAlreadyConfiguredMdevParents() (map[string]struct{
 	return configuredPCICards, nil
 }
 
-func (m *MDEVTypesManager) updateMDEVTypesConfiguration(desiredTypesList []string) (bool, error) {
+func (m *MDEVTypesManager) updateMDEVTypesConfiguration(desiredTypesList []string, doNotRemoveTypesMap map[string]struct{}) (bool, error) {
 	m.mdevsConfigurationMutex.Lock()
 	defer m.mdevsConfigurationMutex.Unlock()
 
 	// construct a map of desired types for lookup
 	desiredTypesMap := make(map[string]struct{})
 	for _, mdevType := range desiredTypesList {
 		desiredTypesMap[mdevType] = struct{}{}
+		doNotRemoveTypesMap[mdevType] = struct{}{}
 	}
-	removeUndesiredMDEVs(desiredTypesMap)
+
+	removeUndesiredMDEVs(doNotRemoveTypesMap)
```

Comment:
This function is hard to read, and this call is particularly confusing (`remove(doNotRemove)`).
It's probably worth adding to the comment above to describe what's happening.